pkg/daemon/daemon.go

```diff
@@ -144,128 +291,41 @@ func (d *daemon) AddPeriodicUpdate() {
 			continue
 		}
 
-		netAttInfo, err := d.kubeClient.GetNetworkAttachmentDefinition(networkNamespace, networkName)
+		networkName, ibCniSpec, err := d.GetIbSriovNetwork(networkID)
 		if err != nil {
-			log.Warn().Msgf("failed to get networkName attachment %s with error: %v", networkName, err)
-			// skip failed networks
-			continue
-		}
-
-		log.Debug().Msgf("networkName attachment %v", netAttInfo)
-		networkSpec := make(map[string]interface{})
-		err = json.Unmarshal([]byte(netAttInfo.Spec.Config), &networkSpec)
-		if err != nil {
-			log.Warn().Msgf("failed to parse networkName attachment %s with error: %v", networkName, err)
-			// skip failed networks
-			continue
-		}
-		log.Debug().Msgf("networkName attachment spec %+v", networkSpec)
-
-		ibCniSpec, err := utils.GetIbSriovCniFromNetwork(networkSpec)
-		if err != nil {
-			addMap.UnSafeRemove(networkID)
-			log.Warn().Msgf("failed to get InfiniBand SR-IOV CNI spec from network attachment %+v, with error %v",
-				networkSpec, err)
-			// skip failed network
+			if strings.Contains(err.Error(), "SR-IOV CNI spec") {
```

Comment:
we can move that to GetIbSriovNetwork
Also why this check is in if err != nil Line 295

Reply:
function GetIbSriovNetwork used for Delete periodic update as well, but remove network from map for Add only